server/authz/rbac.go

```diff
@@ -20,16 +20,28 @@ import (
 	_ "embed"
 	"fmt"
 	"path"
+	"strings"
 
 	"github.com/casbin/casbin/v2"
 	"github.com/casbin/casbin/v2/model"
 	fileadapter "github.com/casbin/casbin/v2/persist/file-adapter"
+	"github.com/fsnotify/fsnotify"
 	"github.com/gin-gonic/gin"
+	"github.com/spf13/viper"
+	"k8s.io/utils/strings/slices"
+
+	"github.com/numaproj/numaflow/pkg/shared/logging"
+	"github.com/numaproj/numaflow/server/authn"
 )
 
 var (
 	//go:embed rbac-model.conf
-	rbacModel string
+	rbacModel     string
+	logger        = logging.NewLogger()
+	userPermCount map[string]int
+	currentScopes []string
+	policyDefault string
+	configReader  *viper.Viper
```

Comment:
Can we make them instance variables belonging to the casbin struct instead of global? They are data that are specific to the object and are used across multiple methods inside the struct. Having too many global variables can make code harder to maintain and understand.

Reply:
Moved to Casbin Struct